eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/RSMergeWriter.java

```diff
@@ -192,6 +193,18 @@ public void writeRSMerge(SubmittedVariantOperationEntity currentOperation)
                 getMergeDestinationAndMergees(mergeCandidates);
         ClusteredVariantEntity mergeDestination = mergeDestinationAndMergees.getLeft();
 
+        // check if any variant with same hash as mergeDestination already exist in DB
+        ClusteredVariantEntity existingClusteredVariantEntity = getClusteredVariantEntityWithHash(mergeDestination);
+        if (existingClusteredVariantEntity != null && existingClusteredVariantEntity.getAccession()!=mergeDestination.getAccession()) {
+            if (mergeDestination.getAccession() == ClusteredVariantMergingPolicy.prioritise(mergeDestination.getAccession(),
+                    existingClusteredVariantEntity.getAccession()).accessionToKeep) {
+                merge(mergeDestination, existingClusteredVariantEntity, currentOperation);
```

Comment:
The `existingClusteredVariantEntity` is also part of the `mergees` so we will likely have to remove it from the list